### PR TITLE
Prune some duplicated dependencies

### PR DESCRIPTION
**Pull request by koushiro**
#  What have you changed?

prune some duplicated dependencies

###  What is the type of the changes?

- Engineering

###  How is the PR tested?

- Manual test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

no

###  Does this PR affect `tidb-ansible`?

no

### Refer to a related PR or issue link (optional)

#5595

### CODE REVIEW

**Comment by koushiro** [comment]
Once `fail` releases a new version, the `rand 0.6` can be completely removed from the dependency graph.

File: components/backup/Cargo.toml [hunk 1]
```diff
@@ -58,6 +58,7 @@ hex = "0.3"
 keys = { path = "../keys" }
```
Comment (5kbpers) [inline components/backup/Cargo.toml]: Don't forget update `hex` here.

Comment (koushiro) [reply components/backup/Cargo.toml]: Oops, I forgot it :rofl:

**Pull request merged**